src/Compilers/VisualBasic/Portable/Emit/SynthesizedPrivateImplementationDetailsSharedConstructor.vb

```diff
@@ -0,0 +1,112 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Collections.Generic
+Imports System.Collections.Immutable
+Imports System.Diagnostics
+Imports System.Linq
+Imports Microsoft.CodeAnalysis.CodeGen
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
+    Friend NotInheritable Class SynthesizedPrivateImplementationDetailsSharedConstructor
+        Inherits SynthesizedGlobalMethodBase
```

Comment:
I started with a base type of SynthesizedConstructorBase. That led to a succession of problems because that type assumes, for example, that the constructor has a valid containing type. After fixing about a half-dozen issues with various hacks, I gave up and switched to SynthesizedGlobalMethodBase. One can see the earlier attempts in the early commits in this PR.